### ISSUE TITLE
Support MQTT persistent session

### ISSUE THREAD

**Issue by ytwig** [aws/aws-iot-device-sdk-java#96]
The SDK doesn't seem to allow setting the **clearSession** flag, as part of the connection, and is hardcoded to true, making every connection as not supporting persistent session.

Can we please add support for this new capability ? 

Source: https://docs.aws.amazon.com/iot/latest/developerguide/mqtt-persistent-sessions.html

**Comment by bretambrose** [comment]
Are you in a position to upgrade to the v2 version of the library? (https://github.com/awslabs/aws-iot-device-sdk-java-v2)

It's currently in developer preview and supports clean session as seen here:

https://github.com/awslabs/aws-crt-java/blob/master/src/main/java/software/amazon/awssdk/crt/mqtt/MqttConnection.java#L223

The v1 iot SDKs are in maintenance mode and it is unlikely any significant new functionality will be added to them.

**Comment by ytwig** [comment]
Thanks for the quick reply.
We are at the starting point of the project, so technically yeah. 
Is v2 production ready?

**Comment by bretambrose** [comment]
It is already being used in a production capacity, but since it's still developer preview, there may be (minor) breaking changes with new releases until GA.

As a follow up to your original question, I discovered that clean session support has been added to the v1 SDK but there hasn't been a release made since it was added.  We are aiming to have a new release out this week, so if you're more comfortable with the v1 SDK, it should be available soon.
